hs_core/hydroshare/resource.py

```diff
@@ -1001,54 +924,66 @@ def publish_resource(user, pk):
                               "metadata or content files or this resource type is not allowed "
                               "for publication.")
 
-    # append pending to the doi field to indicate DOI is not activated yet. Upon successful
-    # activation, "pending" will be removed from DOI field
-    resource.doi = get_resource_doi(pk, 'pending')
+    if istorage.exists(res_coll):
+        bag_modified = istorage.getAVU(res_coll, 'bag_modified')
+        if bag_modified.lower() == "true":
```

Comment:
Add the condition to make sure if bag_modified AVU is None, you also recreate the bag. Check out the django_irods download view function for reference.